src/plots/cartesian/layout_attributes.js

```diff
@@ -211,6 +213,23 @@ module.exports = {
             'and *right* for x axes, and *top*, *middle* (default), and *bottom* for y axes.'
         ].join(' ')
     },
+    matches: {
+        valType: 'enumerated',
+        values: [
+            constants.idRegex.x.toString(),
+            constants.idRegex.y.toString()
+        ],
+        role: 'info',
+        editType: 'calc',
+        description: [
+            'If set to another axis id (e.g. `x2`, `y`), the range of this axis',
+            'will match the range of the corresponding axis in data-coordinates space.',
+            'Moreover, matching axes share auto-range values, category lists and',
+            'histogram auto-bins.',
+            'Note that setting axes simultaneously in both a `scaleanchor` and a `matches` constraint',
+            'is currently forbidden.'
```

Comment:
Is it obvious that for axes to match, they need to have the same type? If it isn't, maybe we should mention it here. On a related note, should we fire `lib.warn` when that happens?

Reply:
> Is it obvious that for axes to match, they need to have the same type?

I'll add it to the description. Thanks!




> . On a related note, should we fire `lib.warn` when that happens?

It should fall into this 

https://github.com/plotly/plotly.js/blob/91431ec425052d0a5ee90182b08adafc6c53cd15/src/plots/cartesian/constraints.js#L70-L72

`Lib.warn` call.

Reply:
> I'll add it to the description.

-> https://github.com/plotly/plotly.js/pull/3506/commits/772efe519828e6efdb436e4dfbb8a6275b230bdf